lib/consoletest.pm

```diff
@@ -49,13 +49,23 @@ Method executed when run() finishes and the module has result => 'fail'
 
 =cut
 sub post_fail_hook {
-    my ($self) = shift;
-    select_console('log-console');
-    $self->SUPER::post_fail_hook;
-    $self->remount_tmp_if_ro;
-    $self->export_logs_basic;
-    # Export extra log after failure for further check gdm issue 1127317, also poo#45236 used for tracking action on Openqa
-    $self->export_logs_desktop;
+    # On Public Cloud everything is done via SSH so the code below doesn't work
+    # TODO: Make some reasonable base post_fail_hook for Public Cloud
+    if (!get_var('PUBLIC_CLOUD')) {
+        my ($self) = shift;
+        select_console('log-console');
+        $self->SUPER::post_fail_hook;
+        $self->remount_tmp_if_ro;
+        $self->export_logs_basic;
+        # Export extra log after failure for further check gdm issue 1127317, also poo#45236 used for tracking action on Openqa
+        $self->export_logs_desktop;
+    }
+}
+
+sub test_flags {
+    if(get_var('PUBLIC_CLOUD')) {
+        return {no_rollback => 1};
+    }
```

Comment:
```suggestion
    return { no_rollback => } if (get_var('PUBLIC_CLOUD'));
    return {};
```
otherwise this implementation broke https://github.com/os-autoinst/os-autoinst/blob/master/autotest.pm#L383 and others like this, cause you set flags to `undef`